addons/pos_sale/models/pos_order.py

```diff
@@ -46,6 +46,14 @@ def _prepare_invoice_vals(self):
                 invoice_vals['partner_id'] = sale_orders[0].partner_invoice_id.id
         return invoice_vals
 
+    def action_pos_order_paid(self):
+        res = super().action_pos_order_paid()
+        if any(p.payment_method_id._is_online_payment() for p in self.payment_ids):
+            sale_orders = self.lines.mapped('sale_order_origin_id')
+            for so in set(sale_orders.filtered(lambda s: s.state in ('draft', 'sent'))):
+                so.action_confirm()
+        return res
```

Comment:
Couldn't you make something like
```suggestion
    def action_pos_order_paid(self):
        res = super().action_pos_order_paid()
        sale_orders = self.lines.mapped('sale_order_origin_id')
        for so in set(sale_orders.filtered(lambda s: s.state in ('draft', 'sent'))):
            so.action_confirm()
        return res
```

And remove the same part in `sync_from_ui` since, when doing `sync_from_ui`, the code will go through `_process_order` and then `_process_saved_order` and if the state is `paid` it will go in `action_pos_order_paid`. This would solve the duplication of the code and the bug we had at first. Wdyt ?